filter.cc

```diff
@@ -96,27 +103,41 @@ Filter::Filter(const Filter &other) {
     data = other.data;
     domainList = other.domainList;
     host = other.host;
+    ruleDefinition = other.ruleDefinition;
+    ruleDefinitionLen = other.ruleDefinitionLen;
   } else {
     if (other.data) {
       data = new char[dataLen];
       memcpy(data, other.data, dataLen);
     } else {
       data = nullptr;
     }
+
     if (other.domainList) {
        size_t len = strlen(other.domainList) + 1;
        domainList = new char[len];
        snprintf(domainList, len, "%s", other.domainList);
     } else {
       domainList = nullptr;
     }
+
     if (other.host) {
       size_t len = strlen(other.host) + 1;
       host = new char[len];
       snprintf(host, len, "%s", other.host);
     } else {
       host = nullptr;
     }
+
+    if (other.ruleDefinition) {
+      size_t len = strlen(other.ruleDefinition) + 1;
+      ruleDefinition = new char[len];
+      snprintf(ruleDefinition, len, "%s", other.ruleDefinition);
+      ruleDefinitionLen = len - 1;
```

Comment:
this is different from above, before the len included the 0 terminated character.